src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,192 @@
+    if (!value.length) {
+      handleDeleteTodo();
+    }
+
+    if (todo.title === value.trim()) {
+      setIsEdit(false);
+
+      return;
+    }
+
+    setValue(text => text.trim());
+    setIsDeleting(true);
+
+    updateTodo({ ...todo, title: value })
+      .then(() => {
+        setIsEdit(false);
+      })
+      .catch(() => {
+        setIsEdit(true);
+        dispatch({
+          type: 'SHOW_ERROR_MESSAGE',
+          payload: { message: 'Unable to update a todo' },
+        });
+      })
+      .finally(() => {
+        setIsDeleting(false);
+      });
+  };
+
+  const updateChecked = (updatedTodo: Todo) => {
+    setIsDeleting(true);
+    updateTodo({ ...todo, completed: !updatedTodo.completed, title: value })
+      .then(() => {
+        dispatch({ type: 'TOGGLE_TODO', payload: { id: todo.id } });
+      })
+      .catch(() => {
+        dispatch({
+          type: 'SHOW_ERROR_MESSAGE',
+          payload: { message: 'Unable to update a todo' },
+        });
+      })
+      .finally(() => {
+        setIsDeleting(false);
+      });
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    handlerBlur();
+  };
+
+  const handleButtonChange = (e: React.KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      setValue(todo.title);
+      setIsEdit(false);
+    }
+  };
+
+  return (
+    <>
+      {isEdit ? (
+        <div
+          data-cy="Todo"
+          className={classNames('todo', { completed: todo.completed })}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              checked={todo.completed}
+              className="todo__status"
+            />
+          </label>
+
+          {/* This form is shown instead of the title and remove button */}
+          <form onSubmit={handleSubmit}>
+            <input
+              ref={inputElem}
+              data-cy="TodoTitleField"
+              type="text"
+              className="todo__title-field"
+              placeholder="Empty todo will be deleted"
+              value={value}
+              onKeyUp={handleButtonChange}
+              onBlur={handlerBlur}
+              onChange={e => setValue(e.target.value)}
+            />
+          </form>
+
+          <div
+            data-cy="TodoLoader"
+            className={classNames('modal', 'overlay', {
+              'is-active': isDeleting,
+            })}
+          >
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ) : (
+        <div
+          data-cy="Todo"
+          className={classNames('todo', { completed: todo.completed })}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
```

Comment:
Use[ destruction assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) for todo